auth/service.cc

```diff
@@ -157,7 +157,7 @@ future<> service::create_keyspace_if_missing(::service::migration_manager& mm) c
     auto& db = _qp.db();
 
     if (!db.has_keyspace(meta::AUTH_KS)) {
-        std::map<sstring, sstring> opts{{"replication_factor", "1"}};
+        std::map<sstring, sstring> opts{{"replication_factor", "3"}};
```

Comment:
This may not work well with clusters with fewer than 3 nodes.
It's also a problem that will happen for any user keyspace that has RF=1, no ?

I see the problem but I think this is not the right way to go. I will try to suggest something

Reply:
I do not even see the problem

Reply:
The problem is before this patch, we allow replace operation with RF = 1, which we lose data. After this patch, we correctly detect that a source node can not be found for keyspace with RF = 1.

Reply:
It is practically the same as disallowing RF=1. With RF=1 there is not HA, so whoever uses it may be well aware of it.

Reply:
As I mentioned in commit log in auth: Increase system_auth from RF 1 to 3
    
    TODO:
    - Handle existing cluster already has system_auth with RF 1.
    - We may need a way to allow replace operation with keyspace with RF 1
      if user allows to lost the data for that keyspace.

The question is  what do we in case there is a RF 1 keyspace and the source node is dead. Before we silently ignore in replacing operation. What should we do now?

Reply:
What's the problem with losing data during a replace with RF=1 ?
I see no other way. The problem I see here is that we're trying to handle auth differently. I am not sure it should be handled differently for the purpose of this patch.

I have been saying for a while that auth should use EverywhereStrategy (which we don't have), to automatically shrink and grow so this becomes automatic. But until that's done, why treat auth differently ? If you had it in RF=1 you lose data upon a lost node, same as you would with any RF=1 keyspace.

Reply:
The question is should we silently make the replace operation succeed with RF = 1 (losing data) or fail the replace operation in this case and ask user to decide what to do. 

For example, user might want to fix the dead node to avoid losing the RF data.

In the new version, I dropped the "auth: Increase system_auth from RF 1 to 3" patch.

Reply:
Asias, I thought about this a bit more and I think I matured my preference now.

If we think about it, the replace operation does not "lose data" when RF=1. Data is already lost. There is no need to fail the replace operation, or warn, or anything.

If we want to be nice and somehow warn users about RF=1, then this should be done at create / alter time. Replace is not the place for it.

The auth thing is a problem indeed because it is a keyspace we create, not the user. Ideally auth would have Everywhere replication but I don't want to make this patch depend on @tgrabiec work on that area.

However as much as this sucks we're making this no worse than before, and our documentation does claim that users using auth should increase the RF. I don't want to trust them to do it, but I don't want to special case anything here either.